compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1218,22 +1218,31 @@ pub fn typeid_for_instance<'tcx>(
                     let trait_id = tcx.fn_trait_kind_to_def_id(closure_args.kind()).unwrap();
                     let tuple_args =
                         tcx.instantiate_bound_regions_with_erased(closure_args.sig()).inputs()[0];
-                    (trait_id, tuple_args)
+                    (trait_id, Some(tuple_args.into()))
                 }
-                ty::Coroutine(..) => (
+                // Implements coroutine, vtable entry could appear in `Coroutine`
+                ty::Coroutine(..) if tcx.is_general_coroutine(instance.def_id()) => (
```

Comment:
Please match on `tcx.coroutine_kind().unwrap`:
* CoroutineKind::Coroutine => Regular
* Otherwise, look at CoroutineDesugaring:
  * CoroutineDesugaring::Async => Future
  * CoroutineDesugaring::AsyncGen => AsyncIterator
  * CoroutineDesugaring::Gen => Iterator